### DISCUSSION TITLE
Documentation about which components need WASM and which use JavaScript

### DISCUSSION THREAD

**Discussion by Ducki** [microsoft/fluentui-blazor#1355] [General]
So by trial and error, I have experienced that components like `FluentSelect` work in a _Server Side Rendered_ scenario, because in order to work it uses Javascript.
On the other hand, `FluentCalendar` has zero interactivity in an SSR scenario and seems to be needed to be loaded within a WASM environment (and specifically needs to be set to `@rendermode="InteractiveWebAssembly"`) in order to select a date and be interactive in general.

Am I just using it wrong or are there indeed differences in implementation regarding the use of JS or WASM?

**Comment by vnbaaij** [comment]
It is not a matter of a component using JavaScript (or not) whether it **fully** works in SSR mode. 

It is all about a component doing **any form of event handling on the Blazor side** that determines if interactivity is needed. If that is the case, then the component (or the page hosting the component or the whole app) needs a rendermode. The rendermode is not restricted to WebAssembly in any way. Can just as well be Server or Auto. 

For example, a `FluentDataGrid` can display perfectly well in SSR mode (see the Weather page in a site created with our Web App template with default settings). It just will not support sorting or resizing as it needs Blazor to do that.

**Reply by Ducki** [reply]
Thank you for the clarification! 😊
